### ISSUE TITLE
channel:foo specification does not work correctly with chroot stores

### ISSUE THREAD

**Issue by lheckemann** [NixOS/nix#1909]
```
% nix build -f channel:nixos-unstable-small --store /tmp hello
[9.4/9.7 MiB DL] downloading 'https://nixos.org/channels/nixos-unstable-small/nixexprs.tar.xz'tar: /nix/store/9w3gxb0j7xwf5291a6nmazr7rm2hvwsn-nixexprs.tar.xz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
[9.7 MiB DL]
error: program 'tar' failed with exit code 2
```

This seems to be because the archive is downloaded into the chroot store, but access is then attempted under the regular nix store path.

**Comment by dtzWill** [comment]
I've seen this in the past, but am no longer able to reproduce (?!).

Oh, right, it looks like this was fixed recently (not in 2.0 release): https://github.com/NixOS/nix/commit/64e486ab63a87b18922bbdb8d2414e74afabb8db

**Comment by lheckemann** [comment]
Good catch, thanks.
